Added Azure Cosmos Connector (#394)
IBM/universal-connectors: filter-plugin/logstash-filter-cosmos-azure-guardium/README.md

	       - Select Review + Create at the bottom of the page.Review the settings and select Create.
	       - After successful creation recently created namespace will appear in resource group.
	
	4. To create event hub :
	       - Go to the Event Hubs Namespace page.
           - Click on '+ Event Hub' to add event hub.
           - Enter unique name for event hub.
           - Choose at least as many partitions as you expect that are required during the peak load  of  your application for that particular event hub.
    	 Ex: If user wants to generate traffic from 2 DB instances then partition count should be at least 2 OR more than that.
           - Click on Review+create.
           - Review the settings and click on Create button.  
	  
    5. Connection string for a event hub:
	       - In the list of event hubs, select your event hub.
	       - On the Event Hubs instance page, from Settings select Shared access policies on the left menu.
	       - In shared access policies click on Add button from top.Give the name to policy and provide permission 'manage' and create the policy.  

> 	       - Select Connection string–primary key string from policy (it would be required in input plugin).

Author: I am using the above one.

			- Stream logs to event hub as mentioned as above.
			- For gmachine , We have to use these consumer group name in input section of configuration file as follows:
				- On one machine in input section of configuration file in consumer group field give name as $Default.
				- On other machine in input section of configuration file in consumer group field give name of other consumer group.
				- Keep all other configurations as it is.

## 4. Connect to Azure Cosmos Database
### Insert/Update data through Data Explorer
1. Login to https://portal.azure.com.
2. Navigate to your Azure Cosmos DB account. Open the Data Explorer pane and then select New Container to create a container.
3. After the container is created, use the Data Structures browser to find and open it. For example, a container with the name Families has been created in the image below.
4. In  Families container, you will see the Items option right below the name of the container. Open this option and you'll see a button, in the menu bar in center of the screen, to create a 'New Item'. You will use this feature to create the JSON items.

### Query Execution through Data Explorer
1. Select New SQL Query.

> 2. Write the SQL query and select Execute Query.

Reviewer: ```suggestion
2. Write the SQL query and click **Execute Query**.
```

## 5. Limitations
1. The following important fields couldn't be mapped with Azure-Cosmos audit logs:
   - Source program : field is left blank, as this information is not embedded in the messages pulled from Azure.

>    - ServerIP : field is populated with 0.0.0.0, as this information is not embedded in the messages pulled from Azure.

Reviewer: ```suggestion
   - ServerIP : this field is populated with 0.0.0.0, as this information is not embedded in the messages pulled from Azure.
```

   - Client HostName : Not available with logs

>    - dbUser : field is populated with NA, as it is not available in ControlPlaneRequest, QueryRuntimeStatistics logs.

Reviewer: ```suggestion
   - dbUser : this field is populated with NA, as it is not available in ControlPlaneRequest, QueryRuntimeStatistics logs.
```

>    - Exception > sqlString : field is populated with NA, as it is not available in Error logs.

Reviewer: ```suggestion
   - Exception > sqlString : this field is populated with NA, as it is not available in Error logs.
```

> 2. Logs will not capture for Failed login attempts.

Reviewer: ```suggestion
2. Logs are not captured for failed login attempts.
```

> 3. For DataPlaneRequest dbUser is NA when AADPrincipalId is Empty.

Reviewer: ```suggestion
3. For DataPlaneRequest, dbUser is NA when AADPrincipalId is empty.
```

> 4. Data Explorer and CLI doesn't support Creation, Updation and Deletion for User and Permission Operations; only the REST API does.

Reviewer: ```suggestion
4. Data Explorer and CLI don't support creation, updating, and deletion for user and permission operations; only the REST API does.
```

> 5. REST API doesn't support Creation, Updation and Deletion for Role Definition and Role Assignment Operations; But the Data Explorer and CLI does.

Reviewer: ```suggestion
5. REST API doesn't support creation, updating, and deletion for role definition and role assignment operations; But the Data Explorer and CLI does.
```

> 6. Data Explorer, CLI and REST API doesn't support Trigger execution; It can only be called via programs.

Reviewer: ```suggestion
6. Data Explorer, CLI and REST API don't support trigger execution; It can only be called via programs.
```

## 6. Configuring the Azure-Cosmos filter in Guardium
The Guardium universal connector is the Guardium entry point for native audit logs. The Guardium universal connector identifies and parses the received events, and converts them to a standard Guardium format. The output of the Guardium universal connector is forwarded to the Guardium sniffer on the collector, for policy and auditing enforcements. Configure Guardium to read the native audit logs by customizing the Azure-Cosmos template.

### Before you begin
* Configure the policies you require. See [policies](https://github.com/IBM/universal-connectors/#policies) for more information.
* You must have permission for the S-Tap Management role. The admin user includes this role by default.
* Download the [guardium_logstash-offline-plugin-azure-cosmos.zip](AzureCosmosOverAzureEventHub/guardium_logstash-offline-plugin-azure-cosmos.zip) plug-in.
* Download the plugin filter configuration file [azure_cosmos.conf](azure_cosmos.conf).

### Procedure
1. On the collector, go to Setup > Tools and Views > Configure Universal Connector.

> 2. Enable the connector if it is already disabled, before proceeding uploading of the UC.

Reviewer: ```suggestion
2. Enable the connector if it is disabled.
```

3. Click Upload File and select the offline [guardium_logstash-offline-plugin-azure-cosmos.zip](AzureCosmosOverAzureEventHub/guardium_logstash-offline-plugin-azure-cosmos.zip) plug-in. After it is uploaded, click OK.
4. Click the Plus sign to open the Connector Configuration dialog box.
5. Type a name in the Connector name field.
6. Update the input section to add the details from [azure_cosmos.conf](azure_cosmos.conf) file's input part, omitting the keyword "input{" at the beginning and its corresponding "}" at the end.

>     - Insert the Connection string in storage_connection from the Access Keys present in the Storage account from azure portal.

Reviewer: ```suggestion
    - Insert the Connection string in storage_connection from the Access Keys present in the Storage account from the Azure portal.
```

7. Update the filter section to add the details from [azure_cosmos.conf](azure_cosmos.conf) file's filter part, omitting the keyword "filter{" at the beginning and its corresponding "}" at the end.

> 8. The "type" field should match in the input and filter configuration section. This field should be unique for every individual connector added.

Author: Readme Changes Incorporated
